### PR TITLE
[CELEBORN-11] ShuffleClient supports MapPartition shuffle write

### PR DESCRIPTION
**Pull request by zhongqiangczq**
# [BUG]/[FEATURE] title

### How was this patch tested?

/cc @related-reviewer
@RexXiong @FMX @waitinfuture 
/assign @main-reviewer

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -20,15 +20,14 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.*;
-import java.util.concurrent.Callable;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
```
Comment (leesf) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:23]: useless changes

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 2]
```diff
@@ -47,8 +46,7 @@
 import org.apache.celeborn.common.network.client.RpcResponseCallback;
 import org.apache.celeborn.common.network.client.TransportClient;
 import org.apache.celeborn.common.network.client.TransportClientFactory;
-import org.apache.celeborn.common.network.protocol.PushData;
-import org.apache.celeborn.common.network.protocol.PushMergedData;
+import org.apache.celeborn.common.network.protocol.*;
```
Comment (leesf) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:49]: ditto

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java [hunk 3]
```diff
@@ -172,4 +175,15 @@ public abstract RssInputStream readPartition(
   public abstract boolean unregisterShuffle(String applicationId, int shuffleId, boolean isDriver);
 
   public abstract void shutDown();
+
+  public abstract int pushDataByteBuf(
```
Comment (leesf) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:179]: please add some description to the method.

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java (hunk 1 above)
Comment (zhongqiangczq) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:23]: it's optimized automaticly by code format

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java (hunk 2 above)
Comment (zhongqiangczq) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:49]: it's optimized automaticly by code format

File: client/src/main/java/org/apache/celeborn/client/ShuffleClient.java (hunk 3 above)
Comment (zhongqiangczq) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClient.java:182]: Tks. I have added descriptions

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java (hunk 1 above)
Comment (leesf) [reply client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:23]: the optimized is determined by your idea import config, you should check it and keep align with the codebase.

**Pull request closed without merging**